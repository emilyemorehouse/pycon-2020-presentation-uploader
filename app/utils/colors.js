const colors = {
  yellow: 'rgb(225, 189, 71);',
  crimson: 'rgb(230,34,45)', // Crimson
  magenta: 'rgb(221,17,117)', // Magenta
  modBlue: 'rgb(65,86,166)', // Mod Blue
  cuttleBlue: 'rgb(24,192,221)', // Cuttle Blue
  darkGrey: 'rgb(68,68,68)', // Dark Grey
  mediumGrey: 'rgb(170,170,170)', // Medium Grey
  lightGrey: 'rgb(238,238,238)', // Light Grey
}

colors.primary = colors.modBlue
colors.secondary = colors.yellow
colors.primaryText = colors.darkGrey
colors.background = colors.lightGrey

export default colors
