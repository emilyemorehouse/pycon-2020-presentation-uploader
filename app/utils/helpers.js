import { isPlainObject } from 'lodash'

const ERROR_DEFAULT = 'An error occurred. Please try again.'
const ERROR_NO_INTERNET = 'Request could not be made. Please check your internet connection.'

export const createWithDoc = ({ envName = '', docFunction = () => {}, component = '' }) => {
  let createComponentWithDoc
  if (envName !== 'production') {
    createComponentWithDoc = docFunction(component) // eslint-disable-line global-require
  }
  return createComponentWithDoc || component
}

export const getErrorMessage = err => {
  // - Axios does not return a response object if the service cannot be reached
  if (!err.response) {
    return ERROR_NO_INTERNET
  }

  // - If we do have a response and data object, use the full error object
  if (err.response && err.response.data) {
    try {
      const errorStringOrObject = Object.values(err.response.data)[0]
      if (isPlainObject(errorStringOrObject)) return errorStringOrObject[0]
      return errorStringOrObject
    } catch (error) {
      // eslint-disable-next-line no-console
      console.warn('Could not infer error message from error response object')
    }
  }

  // If a message has not been set, fallback to our default message
  return ERROR_DEFAULT
}
