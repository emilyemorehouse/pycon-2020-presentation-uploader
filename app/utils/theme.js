import colors from './colors'

const theme = {
  global: {
    font: {
      family: 'Roboto Mono',
      size: '16px',
    },
    colors: {
      brand: colors.primary,
      focus: colors.secondary,
    },
  },
  button: {
    border: {
      radius: '4px',
    },
  },
}

export default theme
