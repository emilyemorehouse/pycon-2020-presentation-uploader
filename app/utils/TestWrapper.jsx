/* eslint-disable react/prop-types */
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { IntlProvider } from 'react-intl'
import { Grommet } from 'grommet'

import theme from './theme'

export const TestWrapper = ({ children }) => (
  <IntlProvider locale="en">
    <MemoryRouter initialEntries={['/']}>
      <Grommet theme={theme}>{children}</Grommet>
    </MemoryRouter>
  </IntlProvider>
)
