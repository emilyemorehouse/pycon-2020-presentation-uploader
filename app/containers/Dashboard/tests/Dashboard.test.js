import React from 'react'
import { render } from '@testing-library/react'

// Components
import { TestWrapper } from 'utils/TestWrapper'
import { Dashboard } from '../index'

const renderComponent = (props = {}) =>
  render(
    <TestWrapper>
      <Dashboard {...props} />
    </TestWrapper>,
  )

/**
 *
 * Tests for Dashboard
 *
 */
describe('Dashboard', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
