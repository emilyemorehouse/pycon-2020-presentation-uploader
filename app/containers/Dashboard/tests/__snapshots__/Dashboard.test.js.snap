// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledGrommet-sc-19lkkz7-0 blnwXx"
  >
    <h2
      class="StyledHeading-sc-1rdh4aw-0 jMKEAN"
    >
      Dashboard
    </h2>
    <div
      class="StyledGrid-sc-1wofa1l-0 dXwRwF"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 bquWcN"
      >
        <h4
          class="StyledHeading-sc-1rdh4aw-0 jOzicV"
        >
          Shows
        </h4>
        <header
          class="StyledBox-sc-13pk1d4-0 hZATeQ"
        >
          <span
            class="StyledText-sc-1sadyjn-0 hUokoe Text__StyledText-sc-1qclgcc-0 ifZuor"
          >
            Sort
          </span>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
          />
          <button
            class="StyledButton-sc-323bzc-0 iNqTxf Button__StyledGrommetButton-sc-5cxguj-0 dKGxUf"
            type="button"
          >
            Add New
          </button>
        </header>
        <div
          class="StyledBox-sc-13pk1d4-0 dfDPuq"
        />
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 RIlxa"
      >
        <h4
          class="StyledHeading-sc-1rdh4aw-0 jOzicV"
        >
          To Dos
        </h4>
        <header
          class="StyledBox-sc-13pk1d4-0 hZATeQ"
        >
          <span
            class="StyledText-sc-1sadyjn-0 hUokoe Text__StyledText-sc-1qclgcc-0 ifZuor"
          >
            Sort
          </span>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
          />
          <button
            class="StyledButton-sc-323bzc-0 iNqTxf Button__StyledGrommetButton-sc-5cxguj-0 dKGxUf"
            type="button"
          >
            Add New
          </button>
        </header>
        <div
          class="StyledBox-sc-13pk1d4-0 dfDPuq"
        />
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 fcPjwD"
      >
        <h4
          class="StyledHeading-sc-1rdh4aw-0 jOzicV"
        >
          Artists & Venues
        </h4>
        <header
          class="StyledBox-sc-13pk1d4-0 hZATeQ"
        >
          <span
            class="StyledText-sc-1sadyjn-0 hUokoe Text__StyledText-sc-1qclgcc-0 ifZuor"
          >
            Sort
          </span>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 cxqgXz"
          />
          <button
            class="StyledButton-sc-323bzc-0 iNqTxf Button__StyledGrommetButton-sc-5cxguj-0 dKGxUf"
            type="button"
          >
            Add New
          </button>
        </header>
        <div
          class="StyledBox-sc-13pk1d4-0 dfDPuq"
        />
      </div>
    </div>
  </div>
</div>
`;
